File: PGS/Lv.3/네트워크.cpp
#include <string>
#include <vector>

using namespace std;

bool visited[200];

void dfs(vector<vector<int>> &computers, int n, int now) {
	visited[now] = true;

	for (int i = 0; i < n; i++) {
		if (computers[now][i] && !visited[i]) {
			dfs(computers, n, i);
		}
	}
}

int solution(int n, vector<vector<int>> computers) {
	int answer = 0;

	for (int i = 0; i < n; i++) {
		if (!visited[i]) {
			dfs(computers, n, i);
			answer++;
		}
	}
	return answer;
}

File: PGS/Lv.3/JAVA/네트워크.java
class Solution {
    
    static boolean[] visited;
    
    void dfs(int[][] computers, int now) {
        visited[now] = true;
        
        for(int next = 0; next < computers.length; next++) {
            if(computers[now][next] == 1 && !visited[next]) {
                dfs(computers, next);
            }   
        }
    }
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        
        visited = new boolean[n];
        
        for(int i = 0; i < n; i++) {
            if(!visited[i]) {
                dfs(computers, i);
                answer++;
            }
        }
        
        return answer;
    }
}